src/borrow_tracker/tree_borrows/perms.rs

```diff
@@ -306,18 +346,20 @@ impl Permission {
     /// remove protected parents.
     pub fn can_be_replaced_by_child(self, child: Self) -> bool {
         match (self.inner, child.inner) {
-            // ReservedIM can be replaced by anything, as it allows all
+            // Cell allows all transitions.
+            (Cell, _) => true,
+            // ReservedIM can be replaced by anything besides Cell, as it allows all
             // transitions.
+            (ReservedIM, Cell) => false,
             (ReservedIM, _) => true,
             // Reserved (as parent, where conflictedness does not matter)
-            // can be replaced by all but ReservedIM,
-            // since ReservedIM alone would survive foreign writes
-            (ReservedFrz { .. }, ReservedIM) => false,
+            // can be replaced by all but ReservedIM and Cell,
+            // since ReservedIM and Cell alone would survive foreign writes
+            (ReservedFrz { .. }, ReservedIM) | (ReservedFrz { .. }, Cell) => false,
```

Comment:
This seems to accumulate the "states handled above". Instead of doing that, would it make sense to do something like this?
```rust
// Nothing can be replaced by Cell.
(_, Cell) => false,
// ReservedIM can be replaced by anything (besides Cell, handled above).
(ReservedIM, _) => true,
```
Then we don't have to repeat `Cell ` a few more times below, right?